kernel/src/platform/mpu.rs

```diff
@@ -20,14 +18,41 @@ pub enum ExecutePermission {
     ExecutionNotPermitted = 0b1,
 }
 
+pub struct Region {
```

Comment:
As this is generic, should it have a `valid` bit or something here? Currently this is working out because bit 0 of attributes is the valid bit and writing the register works, but that's a bit of happenstance. Feels like it might make sense for the "generic" Region struct to have an internal sense of whether it's valid?

Actually, no, I take that back. Anywhere where we'd want an "invalid" region, it should probably just be `None` instead. The _real_ question then is why the `empty` constructor needs to exist? Feels like things should just be putting `None` in wherever `empty` is currently being used (it's a bit of a chip-specific assumption that 0's are a valid "empty" case anyway)